arc_solvers/common/common_utils.py

```diff
@@ -0,0 +1,200 @@
+from typing import Dict, Any
```

Comment:
From what I can tell, this whole file is only here because you have to call `update_params` later, and that call is actually pretty trivial. If at all possible, I'd really like to get rid of this thing. Can you replace whatever that one `update_params` call does with some manual code, and remove this whole file?

There are two reasons why I don't want to have it here. One is that there are clear bugs and inconsistencies in it. But another is that AllenNLP config files are already Jsonnet files, and everything you're doing by "transforming" these config files can be done without any extra code, just with Jsonnet.

Reply:
Turns out Jsonnet is only available in AllenNLP master, not yet in any released version. But they are moving to it, so I guess we have to go there.

Reply:
I agree with Dirk here - it looks like you might be able to scratch this completely if you refactor a little how you are passing parameters to the `SimilarityFunction` in your constructor. This would make the model considerably cleaner.

Reply:
I agree that since this will not be extended, it is an overkill. I will pass the params only in the config file, without inference of the size.

Reply:
Instead of inferring the size in the constructor, you can check that they match and raise an appropriate error if they don't. I think that would be cleaner 👍